mosquitto/readme.md

```diff
@@ -0,0 +1,133 @@
+# Using the Mosquitto library
+
+This samples uses the Mosquitto (MQTT) library to send a message to an Azure IoT Hub.
+
+## Windows prerequisites
+
+1. Follow the [general prerequisites](/README.md#general-rerequisites)
+1. Install [Visual Studio](https://visualstudio.microsoft.com/downloads)
+ 
+    > **NOTE**
+    > Enable the `Desktop development with C++` workload
+
+1. Install [CMake](https://cmake.org/download/)
+
+    > **NOTE**
+    > Enable the `Add CMake to the system PATH for all users` option
+
+1. Install the **x64 version** of [Mosquitto](https://mosquitto.org/download/)
+
+## Linux prerequisites
+
+1. Follow the [general prerequisites](/README.md#general-rerequisites)
+1. Install the build tools:
+
+    ```Shell
+   sudo apt install cmake g++
+    ```
+
+1. Install Mosquitto client library:
+
+    ```Shell
+    sudo apt install libmosquitto-dev
+    ```
+
+## Build
+
+1. Change to the `mosquitto` directory in the cloned repository
+1. Edit the `config.h` file and expand the required connection information:
+
+   ```c
+   #define IOTHUBNAME      "{iothub_name}"
+   #define DEVICEID        "{device_id}"
+   #define SAS_TOKEN       "{sas_token}"
```

Comment:
can we include a sample of what are we expecting here? a.k.a. Iot Hub name: is it just the name, or the FQDN?   also, SAS Toker: just the key or the whole thing?